Move creating and triggering events etc

hamishwillee opened the pull request:
Moving (the useful document) https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events out from the "events guide" under the [Event Reference](https://developer.mozilla.org/en-US/docs/Web/Events). The redirect is automatic, but I also manually fixed up a lot of the links. 

In addition, 
- to make the sub documents visible, I have added a sidebar section which displays the sub pages and other key docs. 
- Deleted sub doc "touchstart" which duplicated equivalent event docs under Element and Document
- Moved sub doc "periodicsync" under ServiceWorkerGlobalScope as that is where is is defined. Have not yet put a link IN ServiceWorkerGlobalScope - I want a sanity check on this. 

There are a couple more comments/questions inline.

hamishwillee committed:
Move creating and triggering events etc

hamishwillee commented on files/en-us/web/api/serviceworkerglobalscope/onperiodicsync/index.html:
```diff
@@ -47,7 +47,7 @@ <h2 id="Specifications">Specifications</h2>
 <h2 id="Browser_compatibility">Browser compatibility</h2>
 
-<p>{{Compat("api.ServiceWorkerGlobalScope.periodicsync")}}</p>
+<p>{{Compat("api.ServiceWorkerGlobalScope.onperiodicsync")}}</p>
```
FYI, this was pointing to wrong item for this attribute

hamishwillee commented on files/en-us/web/api/serviceworkerglobalscope/periodicsync/index.html:
```diff
@@ -70,7 +70,7 @@ <h2 id="Specifications">Specifications</h2>
 <h2 id="Browser_compatibility">Browser compatibility</h2>
 
-<p>{{Compat("api.ServiceWorkerGlobalScope.periodicsync")}}</p>
+<p>{{Compat("api.ServiceWorkerGlobalScope.onperiodicsync")}}</p>
```
Note that this was pointing to the correct attribute, which is not defined. However I think we can reasonably assume symmetry between periodicsync and onperiodicsync. ? 

@ddbeck Is it reasonable to also add `ServiceWorkerGlobalScope.periodicsync` to the BCD and just match up with onperiodicsync. (reason is just docs consistency)

ddbeck replied:
```suggestion
<p>{{Compat("api.ServiceWorkerGlobalScope.periodicsync_event")}}</p>
```

For the `Compat` macro, I'd say don't cover up a missing feature by pointing to a similar feature. I think it's better to have the missing compat data be obviously broken (and have a clear route to fixing it) than to hide what's missing. Instead, point to `api.ServiceWorkerGlobalScope.periodicsync_event`.

(Very tangentially related, I also think the slug for this page is wrong. It also ought to be `periodicsync_event`. More like the [`push` event](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/push_event).)

Typically, there are three or more features to represent an event:

- the event itself, `<eventname>_event`, as a subfeature of the target interface(s) (see [this guideline](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines.md#dom-events-eventname_event)). So, I'd expect some feature called `periodicsync_event`, which we appear to be missing.
- the handler(s), `on<eventname>`, e.g., `onperiodicsync`
- the interface for that event (if applicable), e.g., `PeriodicSyncEvent`

In the boring cases, this data is all the same, but there are fewer boring cases than you'd expect, which is why we have the separate features.

hamishwillee replied:
@ddbeck Thanks! Agreed, and I will rename this doc on Friday when I come back to this (plus do the BCD etc).

hamishwillee replied:
Done. Moved the file/stub, fixed up title, and created BCD

hamishwillee commented on files/en-us/web/events/index/index.html:
```diff
@@ -5,4 +5,4 @@
   - Index
   - events
 ---
-<p>{{Index("/en-US/docs/Web/Events")}}</p>
```
@chrisdavidmills This [Index file](https://developer.mozilla.org/en-US/docs/Web/Events/Index) just lists the other sub pages of events reference. It appears in the new sidebar of event reference page like this

![image](https://user-images.githubusercontent.com/5368500/114512734-c599c980-9c7c-11eb-982e-c0009040322f.png)

And the content (not finished yet) looks a bit like this:
![image](https://user-images.githubusercontent.com/5368500/114512912-f417a480-9c7c-11eb-8c8d-1e6e6d164fa7.png)

My thought was that we don't need the document at all, OR we could just put this info via macro into the Events reference. Any opinion?

chrisdavidmills replied:
@hamishwillee agreed; I don't think we need this.

hamishwillee replied:
Done!

hamishwillee commented on files/en-us/web/events/touchstart/index.html:
```diff
@@ -1,73 +0,0 @@
----
```
FYI Deleted as is duplicate of the Element and Document touchstart event docs

hamishwillee commented on files/en-us/web/events/index.html:
```diff
@@ -6,7 +6,7 @@
   - Reference
 ---
-<p><a href="/en-US/docs/Web/API/Document_Object_Model/Events">DOM Events</a> are fired to notify code of "interesting changes" that may affect code execution. These can arise from user interactions such as using a mouse or resizing a window, changes in the state of the underlying environment (e.g. low battery or media events from the operating system), and other causes.</p>
+<p><a href="/en-US/docs/Learn/JavaScript/Building_blocks/Events">Events</a> are fired to notify code of "interesting changes" that may affect code execution. These can arise from user interactions such as using a mouse or resizing a window, changes in the state of the underlying environment (e.g. low battery or media events from the operating system), and other causes.</p>
```
@chrisdavidmills There are a few places that were linking to https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Events . This is like a minimal summary covering just how to register for/handle events. The learn doc you created is generally better so I changed links to that.

Now this docs is not in our list. Was wondering whether you see value in keeping it. I kind of like the "just how I register for events" sub topic. However I could delete it and create a redirect. Thoughts?

chrisdavidmills replied:
@hamishwillee I think we hatched a sufficient plan for what to do with this in our call — keep it under the events reference as a minimal reference guide for professionals that want a quick recap; defer to my events learning guide for beginners who need mroe context.

hamishwillee replied:
Agreed. Added to list. Will do as a separate action.

hamishwillee commented:
@peterbe I'm getting redirect issues from CI again. 
```
$ env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js validate-redirects en-us --strict
removing orphaned redirect (to doesn't exists): /en-US/docs/DOM/Mozilla_event_reference/touchstart	/en-US/docs/Web/Events/touchstart
removing orphaned redirect (to doesn't exists): /en-US/docs/Mozilla_event_reference/touchstart	/en-US/docs/Web/Events/touchstart
removing orphaned redirect (to doesn't exists): /en-US/docs/Web/Reference/Events/touchstart	/en-US/docs/Web/Events/touchstart
info: _redirects.txt for en-us is causing issues: Error:  _redirects.txt for en-us is flawed
```
This case looks like it might be due to deletion of  `/en-US/docs/Web/Events/touchstart` (this is duplicate of same event under Document and Element). Does that seem likely to you? However I'm not sure because the warning above is telling me that those problems are fixed (ie "removing orphaned redirect").

Advice appreciated (note, I did deletion with tools).

chrisdavidmills reviewed (approved):
@hamishwillee I've had a look, and I think this all makes sense. Is there anything else you wanted to add here? If not, I'll get it merged.

peterbe commented:
> @peterbe I'm getting redirect issues from CI again.
> 
> ```
> $ env-cmd --silent cross-env CONTENT_ROOT=files node node_modules/@mdn/yari/tool/cli.js validate-redirects en-us --strict
> removing orphaned redirect (to doesn't exists): /en-US/docs/DOM/Mozilla_event_reference/touchstart	/en-US/docs/Web/Events/touchstart
> removing orphaned redirect (to doesn't exists): /en-US/docs/Mozilla_event_reference/touchstart	/en-US/docs/Web/Events/touchstart
> removing orphaned redirect (to doesn't exists): /en-US/docs/Web/Reference/Events/touchstart	/en-US/docs/Web/Events/touchstart
> info: _redirects.txt for en-us is causing issues: Error:  _redirects.txt for en-us is flawed
> ```
> 
> This case looks like it might be due to deletion of `/en-US/docs/Web/Events/touchstart` (this is duplicate of same event under Document and Element). Does that seem likely to you? However I'm not sure because the warning above is telling me that those problems are fixed (ie "removing orphaned redirect").
> 
> Advice appreciated (note, I did deletion with yarn).

The message "removing orphaned redirect (to doesn't exists)" is confusing and misleading. The problem is that your destination URL doesn't exist. So it should say that, but it doesn't :(
You can't have a to-URL that (that doesn't start with `http`) that doesn't resolve to a file. But it's misleading you with the confusing label. 

You'd need to remove these lines from the `_redirects.txt` file manually I suspect.

hamishwillee committed:
Fix up slug and title

hamishwillee committed:
Move periodicsync event to correct location/stub

hamishwillee committed:
Add redirect for moved file (normal yarn content move did nothing)

hamishwillee committed:
Delete the Events index (redundant)

hamishwillee committed:
Fix up redirect to direct to one of the touch event pages

hamishwillee commented:
@chrisdavidmills This should be good as far as it goes now. I will self merge. Changes since you last looked were fixes to the redirects, deletion of index file (as discussed/agreed). I also moved the periodicsync event to the correct location and updated it's title as suggested by Daniel
